runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingSpotOverlay.java

```diff
@@ -123,37 +158,139 @@ public Dimension render(Graphics2D graphics)
 
 			if (config.showSpotTiles())
 			{
-				Polygon poly = npc.getCanvasTilePoly();
+				// each entry in the list corresponds to the same tile
+				Polygon poly = firstSpot.getCanvasTilePoly();
 				if (poly != null)
 				{
 					OverlayUtil.renderPolygon(graphics, poly, color.darker());
 				}
 			}
 
-			if (config.showSpotIcons())
+			for (int i = 0; i < fishingSpots.getSize(); i++)
 			{
-				BufferedImage fishImage = itemManager.getImage(spot.getFishSpriteId());;
-				if (fishImage != null)
+				FishingSpot spot = fishingSpots.getSpot(i);
+				NPC npc = fishingSpots.getNpc(i);
+
+				int spotCount = fishingSpots.getCount(spot);
+
+				if (config.showSpotIcons())
 				{
-					Point imageLocation = npc.getCanvasImageLocation(fishImage, npc.getLogicalHeight());
-					if (imageLocation != null)
+					BufferedImage fishImage = null;
+
+					if (spotCount > 1 && !(config.showSpotNames()))
 					{
-						OverlayUtil.renderImageLocation(graphics, imageLocation, fishImage);
+						fishImage = itemManager.getImage(spot.getFishSpriteId(), spotCount, true);
+					}
+					else
+					{
+						fishImage = itemManager.getImage(spot.getFishSpriteId());
+					}
+
+					if (fishImage != null)
+					{
+						Point imageLocation = npc.getCanvasImageLocation(fishImage, 34);
+						if (imageLocation != null)
+						{
+							int offset = (i * 34) - ((34 * (fishingSpots.getSize() - 1)) / 2);
+							Point shiftedImageLocation = new Point(imageLocation.getX() + offset, imageLocation.getY());
+							OverlayUtil.renderImageLocation(graphics, shiftedImageLocation, fishImage);
+						}
+					}
+				}
+
+				if (config.showSpotNames())
+				{
+					String text = spot.getName();
+					if (spotCount > 1)
+					{
+						text += " (" +  spotCount + ")";
+					}
+					Point textLocation = npc.getCanvasTextLocation(graphics, text, 74 + (i * 20));
+					if (textLocation != null)
+					{
+						OverlayUtil.renderTextLocation(graphics, textLocation, text, color.darker());
 					}
 				}
 			}
+		}
+
+		return null;
+	}
 
-			if (config.showSpotNames())
+	private class FishingSpotInfo
```

Comment:
Move this to it's own file

Reply:
I can certainly do that. In regards to design, I assume you recommend moving the class to its own file because it contains more than basic getters/setters?

Reply:
If you can, get on the RuneLite discord and talk with the guys in the #development chat channel. It sounds like they would like different/ smaller solution to this, and I wouldn't want you to waste your time.

As far as the separate file, I think it is just convention with this project to keep everything separate. The purpose for that might be to enforce modularity and make things obvious for other developers, but don't quote me on it.